pkg/models/gacha.go

```diff
@@ -136,33 +135,39 @@ func GetAllCharacters() ([]Character, error) {
 	}
 	return characters, nil
 }
-func (uc *UserCharacter) CreateUserCharacter() *UserCharacter {
-	db.Create(&uc)
-	return uc
-}
+func CreateUserCharacterBatch(userCharacters []*UserCharacter) error {
+	db := config.GetDB()
+	tx := db.Begin()
+	batchSize := 1000 // Adjust the batch size as per your requirements
 
-func DrawCharacter(characters []Character, characterPool []Character) (*Character, error) {
+	for i := 0; i < len(userCharacters); i += batchSize {
+		end := i + batchSize
+		if end > len(userCharacters) {
+			end = len(userCharacters)
+		}
 
-	if len(characterPool) == 0 {
-		return nil, errors.New("empty character pool")
-	}
-	randIndex, err := rand.Int(rand.Reader, big.NewInt(int64(len(characterPool))))
-	if err != nil {
-		return nil, err
+		batch := userCharacters[i:end]
+		if err := createBatch(tx, batch); err != nil {
+			tx.Rollback()
+			return err
+		}
 	}
-	index := int(randIndex.Int64())
-	selectedCharacter := &characterPool[index]
 
-	// Save the gacha result in the database
-	gachaResult := GachaResult{
-		CharacterID:   selectedCharacter.ID,
-		CharacterName: selectedCharacter.Name,
-	}
-	if err := gachaResult.SaveGachaResult(); err != nil {
-		return nil, err
+	if err := tx.Commit().Error; err != nil {
+		tx.Rollback()
+		return err
 	}
 
-	return selectedCharacter, nil
+	return nil
+}
+
+func createBatch(tx *gorm.DB, batch []*UserCharacter) error {
```

Comment:
This function name makes it difficult to understand what is being done from the outside.

Reply:
Okay I will change the function name.